source/fe/mapping_cartesian.cc

```diff
@@ -315,6 +315,59 @@ MappingCartesian<dim, spacedim>::maybe_update_normal_vectors(
 
 
 
+template <int dim, int spacedim>
+void
+MappingCartesian<dim, spacedim>::maybe_update_jacobian_derivatives(
+  const InternalData &             data,
+  const CellSimilarity::Similarity cell_similarity,
+  internal::FEValuesImplementation::MappingRelatedData<dim, spacedim>
+    &output_data) const
+{
+  if (data.update_each & update_jacobian_grads)
+    if (cell_similarity != CellSimilarity::translation)
+      for (unsigned int i = 0; i < output_data.jacobian_grads.size(); ++i)
+        output_data.jacobian_grads[i] = DerivativeForm<2, dim, spacedim>();
```

Comment:
Hm, on second thought, I think this may be the best you can do. The `MappingCartesian` class doesn't own the `MappingRelatedData` (it's owned by the `FEValues` object) and we can't initialize it any earlier than when we see it the first time. I think there is nothing else you can do.

By the way, `cell_similarity != CellSimilarity::translation` also on the very first cell one encounters.